src/ProjectTemplates/Web.ProjectTemplates/content/Api-CSharp/Program.Main.cs

```diff
@@ -0,0 +1,44 @@
+using System.Text.Json.Serialization;
+#if NativeAot
+using Microsoft.AspNetCore.Http.Json;
+using Microsoft.Extensions.Options;
+#endif
+namespace Company.ApiApplication1;
+
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        var builder = WebApplication.CreateBuilder(args);
+        builder.Logging.AddConsole();
+
+        #if (NativeAot)
+        builder.Services.ConfigureHttpJsonOptions(options =>
+        {
+            options.SerializerOptions.AddContext<AppJsonSerializerContext>();
+        });
+
+        #endif
+        var app = builder.Build();
+
+        var sampleAlbums = AlbumGenerator.GenerateAlbums().ToArray();
+
+        var api = app.MapGroup("/albums");
+        api.MapGet("/", () => sampleAlbums);
+        api.MapGet("/{id}", (int id) =>
+            sampleAlbums.FirstOrDefault(a => a.Id == id) is { } album
+                ? Results.Ok(album)
+                : Results.NotFound($"Album with id {id} not found"));
+
+        app.Run();
+    }
+}
+
+#if (NativeAot)
+[JsonSerializable(typeof(string))]
+[JsonSerializable(typeof(Album))]
+[JsonSerializable(typeof(Album[]))]
+internal partial class AppJsonSerializerContext : JsonSerializerContext
```

Comment:
Should we put this in a separate AppJsonSerializerContext.cs file? Is it easy enough to conditionally include files to the template?

Reply:
I'm in two minds right now and have made all the logical arguments in my head both ways, but landed here aesthetically.  Interested to hear others' thoughts.

Reply:
One part of me thinks: "Put each top level type in its own file", because that's the way I would do it. But the way you've proposed it here makes sense as well. Having this tiny little class in its own file seems going to far.